src/Stratis.Bitcoin/P2P/PeerConnectorDiscovery.cs

```diff
@@ -57,8 +57,16 @@ public override PeerAddress FindPeerToConnectTo()
 
             PeerAddress peer = null;
 
-            while (!this.nodeLifetime.ApplicationStopping.IsCancellationRequested && peerSelectionFailed < 50)
+            while (!this.nodeLifetime.ApplicationStopping.IsCancellationRequested)
             {
+                if (peerSelectionFailed > 50)
```

Comment:
I think we should put a wait here

Reply:
Yeah it could be beneficial.... How long?

Reply:
Maybe put it after each failed attempt?